src/renderer/shared/lib/utils/balance.ts

```diff
@@ -188,34 +188,36 @@ export const totalAmount = <T extends AssetBalance>(balance?: T): string => {
   return bnFree.add(bnReserved).toString();
 };
 
-export const lockedAmountBN = ({ locked = [] }: Balance) => {
-  const bnLocks = locked.map((lock) => new BN(lock.amount));
-  const bnFrozen = bnLocks.reduce((acc, bnLock) => acc.add(bnLock), new BN(0));
+export const lockedAmountBN = (balance: Balance): BN => {
+  if (!balance.locked) return BN_ZERO;
 
-  return bnFrozen;
+  return balance.locked.reduce((acc, lock) => acc.add(lock.amount), BN_ZERO);
 };
 
 export const lockedAmount = (balance: Balance): string => {
   return lockedAmountBN(balance).toString();
 };
 
 export const transferableAmountBN = <T extends AssetBalance>(balance?: T): BN => {
-  if (!balance) return BN_ZERO;
+  if (!balance?.free || !balance?.frozen || !balance?.reserved) return BN_ZERO;
 
-  const bnFree = new BN(balance.free || ZERO_BALANCE);
-  const bnFrozen = new BN(balance.frozen || ZERO_BALANCE);
+  const { free, frozen, reserved } = balance;
+
+  const freeCannotDropBelow = frozen.gt(reserved) ? frozen.sub(reserved) : BN_ZERO;
 
-  return bnFree.gt(bnFrozen) ? bnFree.sub(bnFrozen) : BN_ZERO;
+  return free.gt(freeCannotDropBelow) ? free.sub(freeCannotDropBelow) : BN_ZERO;
 };
 
 export const transferableAmount = <T extends AssetBalance>(balance?: T): string => {
   return transferableAmountBN(balance).toString();
 };
 
-export const stakedAmount = ({ locked = [] }: Balance): string => {
-  const bnLocks = locked.find((lock) => lock.type === LockTypes.STAKING);
+export const stakedAmount = (balance: Balance): string => {
+  if (!balance.locked) return ZERO_BALANCE;
+
+  const bnLocks = balance.locked.find((lock) => lock.type === '0x7374616b696e6720');
```

Comment:
Are you sure about using strings here? Looks like it decrease readability. Enums looks better here for me.